dev/com.ibm.ws.security.wim.registry/src/com/ibm/ws/security/wim/registry/util/DisplayNameBridge.java

```diff
@@ -206,28 +206,37 @@ public String getUserDisplayName(String inputUserSecurityName) throws EntryNotFo
-                            returnValue = "";
-                        else
-                            returnValue = loginAccount.getDisplayName().get(0);
-                    } else if (mappedProp.equals(SchemaConstants.PROP_PRINCIPAL_NAME) && foundInURBridge) {
-                        String outputUserPrincipalAttr = this.propertyMap.getOutputUserPrincipal(idAndRealm.getRealm());
-                        if (!this.mappingUtils.isIdentifierTypeProperty(outputUserPrincipalAttr)) {
-                            returnValue = (String) loginAccount.get(outputUserPrincipalAttr);
+                // This check might cause inconsistencies for customers whose
+                // filters don't match what has been set in the entity objectclasses.
+                // To fix this, they need to add the objectclass to the entity
+                if (returnList.get(0) instanceof PersonAccount) {
+                    PersonAccount loginAccount = (PersonAccount) returnList.get(0);
+                    // f113366
+                    if (!this.mappingUtils.isIdentifierTypeProperty(outputAttrName)) {
+                        //returnValue = loginAccount.getString(this.propertyMap.getOutputUserDisplayName(idAndRealm.getRealm()));
+                        String mappedProp = outputAttrName;
+                        if (mappedProp.equals("displayName")) {
+                            if (loginAccount.getDisplayName().size() == 0)
+                                returnValue = "";
+                            else
+                                returnValue = loginAccount.getDisplayName().get(0);
+                        } else if (mappedProp.equals(SchemaConstants.PROP_PRINCIPAL_NAME) && foundInURBridge) {
+                            String outputUserPrincipalAttr = this.propertyMap.getOutputUserPrincipal(idAndRealm.getRealm());
+                            if (!this.mappingUtils.isIdentifierTypeProperty(outputUserPrincipalAttr)) {
+                                returnValue = (String) loginAccount.get(outputUserPrincipalAttr);
+                            } else {
+                                returnValue = (String) loginAccount.getIdentifier().get(outputUserPrincipalAttr);
+                            }
                         } else {
-                            returnValue = (String) loginAccount.getIdentifier().get(outputUserPrincipalAttr);
+                            returnValue = (String) loginAccount.get(mappedProp);
                         }
                     } else {
-                        returnValue = (String) loginAccount.get(mappedProp);
+                        returnValue = (String) loginAccount.getIdentifier().get(outputAttrName);
                     }
                 } else {
-                    returnValue = (String) loginAccount.getIdentifier().get(outputAttrName);
+                    if (tc.isDebugEnabled()) {
+                        Tr.debug(tc, "Entity found was not of type \"PersonAccount.\"", returnList.get(0));
```

Comment:
Should say something like:

No groups will be returned for user because the entity was not of type "PersonAccount".